File: benchmarks/values/decimal_e.rb
require "bigdecimal"

t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)

e = BigDecimal("0")
rep = 0
while rep < 100000
  e = BigDecimal("0")
  factorial = BigDecimal("1")
  i = 0
  while i <= 100
    e = e + BigDecimal("1") / factorial
    factorial = factorial * (i + 1)
    i += 1
  end
  rep += 1
end

result = (e * 1000000).to_i

t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
puts result
puts "elapsed: #{'%.3f' % (t1 - t0)}s"

File: benchmarks/values/decimal_e.py
import time
from decimal import Decimal, getcontext

getcontext().prec = 50

t0 = time.perf_counter()

e = Decimal("0")
for rep in range(100000):
    e = Decimal("0")
    factorial = Decimal("1")
    for i in range(101):
        e = e + Decimal("1") / factorial
        factorial = factorial * (i + 1)

result = int(e * 1000000)

t1 = time.perf_counter()
print(result)
print(f"elapsed: {t1 - t0:.3f}s")
